### PR TITLE
Loosen our comparison tolerances for floats

### PR DESCRIPTION
**Pull request by cory-signalfx**
# Summary

Use `MaxInt32` for range defaults and value comparisons.

# Motivation

In #66 it was reported that floating point is being floating point. Due to floating point fuckery sometimes we get an unclean plan. I considered switching to `string` but that doesn't fix the `signalfx-go` library which is going to still return 0 defaults.* To that end, this loosens the tolerance of our checks and just ropes off anything above `MaxInt32` or below `-MaxInt32`. This seems safe because `float64` has > 32 bits of mantissa so it can accurately represent this number with no fear of serialization problems.

Thanks, Go!

* We can't make 0 unusable for these values because of color ranges, which heavily use 0.

### CODE REVIEW

**Comment by cory-signalfx** [comment]
Tagging in @keitwb for some Go sanity checks. Any better ideas?

File: signalfx/util.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func getColorScaleOptions(d *schema.ResourceData) []*chart.SecondaryVisualizatio
 }
 
 func getValueUsingMaxFloatAsDefault(v float64) *float32 {
-	if v >= math.MaxFloat32 || v <= -math.MaxFloat32 {
```
Comment (keitwb) [inline signalfx/util.go:182]: The function name seems misleading, it doesn't "use max float as default" anywhere in this function.

Comment (keitwb) [inline signalfx/util.go:182]: Also is there any good reason not to just clamp the value to max/min float32 if it is OOB?

Comment (cory-signalfx) [reply signalfx/util.go:182]: Changed!

**Review by keitwb — APPROVED** [review]
LGTM.  Hacky but sane given the limitations of Terraform config.

**Comment by cory-signalfx** [comment]
This should be fixed in #174

**Pull request closed without merging**